api/envoy/config/rbac/v3/rbac.proto

```diff
@@ -66,19 +66,28 @@ message RBAC {
 
   // Should we do safe-list or block-list style access control?
   enum Action {
-    // The policies grant access to principals. The rest is denied. This is safe-list style
+    // The policies grant access to principals. The rest are denied. This is safe-list style
     // access control. This is the default type.
     ALLOW = 0;
 
-    // The policies deny access to principals. The rest is allowed. This is block-list style
+    // The policies deny access to principals. The rest are allowed. This is block-list style
     // access control.
     DENY = 1;
+
+    // The policies set the "envoy.log" dynamic metadata key based on if requests match
+    LOG = 2;
   }
 
-  // The action to take if a policy matches. The request is allowed if and only if:
+  // The action to take if a policy matches.
+  // The request is allowed if and only if:
   //
-  //   * `action` is "ALLOWED" and at least one policy matches
+  //   * `action` is "ALLOW" and at least one policy matches
   //   * `action` is "DENY" and none of the policies match
+  //   * `action` is "LOG"
```

Comment:
make it clear that it log if at least one policy matches.

Reply:
A couple lines below, there's a comment discussing when envoy.log is set

Reply:
I'll make it clearer

Reply:
Better to explicitly call out that when the action is LOG, the request is always allowed. 

I know this is described above but when people looking at the action comment here it is still confusing especially when compared with the other two actions (ALLOW/DENY) that allows or denies based on the policy matching.

Reply:
LOG does not affect allow or deny decision, correct?

Reply:
Exactly, so all requests are allowed.

Reply:
Sorry I'm late to the game here but given that LOG is really an action on top of other actions, should it be a separate boolean of some type? I'm thinking of the future in which we have different action types for WAF such as TARPIT, etc. and it seems like we might want to also log?

Reply:
The difficulty with combining LOG with allow/deny in the same config is that the requests we want to log might be different than those we want to allow and deny, requiring a separate list of policies to match on. I'm not too familiar with the details of the WAF proposal, but wouldn't that require an expanded list of actions anyway past ALLOW/DENY? So an action that sets dynamic metadata for logging could be one of those actions?

Reply:
I'm just refreshing myself on the RBAC proto, but the fact that there is a single top level action seems like it's not going to work very well for other actions, let alone log. I'm still kind of confused how this is going to be used. Can you provide a small contrived example? It seems like you would want to use logging with either ALLOW or DENY type, and eventually potentially have other actions based on matching?

Reply:
Say a user wants to log all GET requests from principal "cluster.local/ns/default/sa/admin". They deploy the RBAC filter with the LOG action, and then any request that matches this policy will result in the metadata key to be set to true. The user can use their custom telemetry extension to check this key to determine whether to log, or this can also be supported natively by adding an access log filter that checks this key.

Reply:
The way the config is currently set up does not allow for multiple actions per filter, meaning users would have to deploy a separate filter for this. Probably the most extensible way to allow for multiple actions right now would be to support multiple RBAC configs under "rules" in the [filter config](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/rbac/v3/rbac.proto#envoy-v3-api-msg-extensions-filters-http-rbac-v3-rbac). This will likely be solved by the WAF filter though.

Reply:
So is your plan to deploy multiple RBAC filters until we sort out multiple actions?

Reply:
Yep, that's right. We can also create support for multiple configs in a filter, since I assume development for WAF is still early stages.

Reply:
OK got it. Alright this seems fine to me then and we can sort out the changes to support multiple actions later.